modules/nf-core/matrixfilter/main.nf

```diff
@@ -0,0 +1,22 @@
+process MATRIXFILTER {
+    tag "$meta"
+    label 'process_medium'
+    conda (params.enable_conda ? "r-base" : null)
+    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
+        'https://depot.galaxyproject.org/singularity/r-base:4.2.1' :
+        'quay.io/biocontainers/r-base:4.2.1' }"
+
+    input:
+    tuple val(meta), path(samplesheet), path(abundance)
+
+    output:
+    tuple val(meta), path("*.filtered.tsv")             , emit: filtered
```

Comment:
This should just be `${prefix}` and specified via `ext.prefix` (should be used as with ext.args), and a check added for no name clash (to force a dev to supply prefix)

Reply:
Not 100% sure what you mean there @jfy133 . I've changed the output to be renamed according to the prefix derived from the meta, feel free to expand a bit and I'll try to match what you're after.

Reply:
Ah, sorry that is indeed unclear. I posted it on the wrong line.

In the R templtae script, you shoud use the module=-local `${prefix}` variable, and therefore here in the module in the `script:` block (like you normally would have for any other module to have the `prefx = task.ext.prefix ?: 'meta.id'` thing.

Does that make sense? Basically I'm trying to push this module back to as close as the module template as possible :)

Reply:
Thanks! Is the [template edit](https://github.com/pinin4fjords/nf-core-modules/blob/f2e2ae71685c7e8a6d153feff8ab71fd597fb948/modules/nf-core/custom/matrixfilter/templates/matrixfilter.R#L165) not fulfilling this? 

I'm intentionally not referring to `meta` explicitly in the template due to previous feedback about meta assumptions ;-). But it's done in the tests from [the nextflow.config](https://github.com/pinin4fjords/nf-core-modules/blob/f2e2ae71685c7e8a6d153feff8ab71fd597fb948/tests/modules/nf-core/custom/matrixfilter/nextflow.config#L17).

Reply:
Again, not that setting `prefix` in the script block won't work here- the template doesn't seem to receive new variable definitions from `script` (not sure if that's a bug or a feature).

Reply:
Oh, schade :( ok!